### ISSUE TITLE
ERROR: XHR failed. Technical details: {"name":"Error","message":"XHR failed"}

### ISSUE THREAD

**Issue by alan-wr** [microsoft/vscode-file-downloader#10]
File Downloader extension now active.
Starting download from https://*****.bz2
ERROR: XHR failed. Technical details: {"name":"Error","message":"XHR failed"}

**Comment by daniv-msft** [comment]
Thanks @alan-wr for reporting this issue. As of today, from what we've seen, this error seems to be related to network issues and the VS Code API we rely on doesn't provide us much more information.
Here is the issue we logged on their side to investigate this: https://github.com/microsoft/vscode/issues/108760

In parallel, we're considering moving to another API to bypass these errors in the future.

Could you please provide more context about this from your side? When do you get this error?

**Comment by alan-wr** [comment]
Thanks @daniv-msft for reply it.

About 
const progressCallback = (downloadedBytes: number, totalBytes: number | undefined)

I found the variable totalBytes does not get real number.

How can I improve it?

**Comment by daniv-msft** [comment]
@alan-wr Yes, this is unfortunately a known issue while we move to the other API I was mentioning above. The present API, provided by VS Code, doesn't give us this information. I expect that we'll switch to the new API in the coming weeks, but that has other implications we are reviewing carefully.

**Comment by alan-wr** [comment]
Many Thanks @daniv-msft for reply it.

So I hope I can get new API ASAP some day.

Thanks your hard work.

**Comment by alan-wr** [comment]
Use file-download to download same URL, does not get XHR failed.
But it not support progress and unzip.

https://www.npmjs.com/package/file-download

````

  var request = req.get(file, function(response) {

    if (response.statusCode === 200) {

      mkdirp(options.directory, function(err) { 
        if (err) throw err
        var file = fs.createWriteStream(path)
        response.pipe(file)
      })

    } else {

      if (callback) callback(response.statusCode)

    }

    response.on("end", function(){
      if (callback) callback(false, path)
    })

    request.setTimeout(options.timeout, function () {
      request.abort()
      callback("Timeout")
    })

  }).on('error', function(e) {

    if (callback) callback(e)

  })
````

**Comment by daniv-msft** [comment]
@alan-wr We just got an update on the issue logged on VS Code side. Getting the network logs would help investigating this further.

Would it be possible for you to:
- Close all VS Code windows.
- Open a command prompt.
- Run `code --log-net-log=<some-absolute-path>/netlog.json` to start VS Code and asking it to record network logs.
- Reproduce the XHR issue you see with our extension.
- Attach your logs to this issue.

That would be helpful to understand more of what goes wrong in this case.

**Comment by abadfish** [comment]
What do you replace <some-absolute-path> with? 
I am having this problem as well. Fresh install on Ubuntu 20

**Comment by chirag127** [comment]
**Corp proxy repro: XHR fails with no override path on Windows 11 + corp HTTPS proxy**

**Environment:**
- OS: Windows 11 Enterprise 10.0.26200
- VS Code 1.122+, Node 24 (extension host)
- Network: corporate HTTPS-intercepting proxy (Zscaler / similar), system proxy auto-detected by VS Code
- `http.proxy` setting in VS Code: not explicitly set (inherits system proxy)

**What happens:**

The extension uses VS Code's `vscode.workspace` HTTP client (the XHR path), which honours `HTTPS_PROXY` / system proxy. On a corp machine the proxy chain looks like:

```
Extension → VS Code XHR → corp proxy (TLS intercept) → internet
```

The corp proxy presents its own cert. For internal package servers (e.g. an Artifactory or internal S3 bucket) where the cert is issued under the corp CA but NOT trusted by the extension host's bundled Node CA store, the XHR fails silently with `{"name":"Error","message":"XHR failed"}` — no `proxyStrictSSL` or `NODE_EXTRA_CA_CERTS` override surface is exposed to extension consumers.

**Specific gap vs existing comments:**

The existing thread covers general XHR failure and network log collection. The corp-proxy scenario is distinct:

1. `--log-net-log` shows the request leaves VS Code correctly, but the proxy returns a TLS error (cert not trusted), logged as `ERR_TUNNEL_CONNECTION_FAILED` or `ERR_CERT_AUTHORITY_INVALID`.
2. There is no way for an extension consumer (or end-user) to pass `NODE_EXTRA_CA_CERTS` or `rejectUnauthorized: false` through the `FileDownloader` API to the underlying XHR call.
3. `"http.proxyStrictSSL": false` in VS Code settings does NOT propagate to the extension host's Electron net module used by XHR.

**Suggested fix / workaround gap:**

Expose an optional `downloadOptions.strictSSL?: boolean` (or respect `vscode.workspace.getConfiguration('http').get('proxyStrictSSL')`) in the `IFileDownloader.downloadFile()` API so corp users can disable strict SSL for downloads behind intercepting proxies — matching how VS Code's built-in extension gallery handles the same setting.

**Workaround (partial):** Set `"http.proxySupport": "fallback"` and add the corp CA cert to the system trust store. This works only if the corp CA is importable (some locked-down environments block this).
